### PR TITLE
Set FIREBASE_PROJECT env variable when emulating functions

### PR DESCRIPTION
**Pull request by laurenzlong**
This PR ensures that emulation of HTTPS will still be successful when there are non-HTTPS functions in index.js, and that admin.initialize(functions.config().firebase) works.

### CODE REVIEW

**Comment by laurenzlong** [comment]
This now fully works to emulate functions.config().firebase

Must be used with this SDK: https://storage.googleapis.com/firebase-preview-drop/node/firebase-functions/firebase-functions-0.5.7-preview1.tgz. 

PR for SDK here: https://github.com/firebase/firebase-functions/pull/51/files

File: lib/parseTriggersEmulator.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+var path = require('path');
+var RSVP = require('rsvp');
+
+var TRIGGER_PARSER = path.resolve(__dirname, './triggerParser.js');
+
+module.exports = function(firebaseConfig, projectId, dbInstance, sourceDir) {
+  return new RSVP.Promise(function(resolve, reject) {
+    var env = {
+      GCLOUD_PROJECT: projectId,
+      DB_NAMESPACE: dbInstance,
```
Comment (mbleigh) [inline lib/parseTriggersEmulator.js]: Where did this come from? Isn't that included in FIREBASE_PROJECT already? Have I just forgotten the trigger parser code?

File: lib/serve/functions.js [hunk 2]
```diff
@@ -75,44 +83,62 @@ function _startEmulator(options) {
...
+  var controllerConfig = _.merge({tail: true}, _getPorts(options));
   controller = new EmulatorController(controllerConfig);
+  var firebaseConfig;
 
-  return controller.start({ detached: false, stdio: ['ignore', process.stdout, process.stderr]}).then(function() {
+  return functionsConfig.getFirebaseConfig(projectId, options.instance)
+  .then(function(result) {
+    firebaseConfig = JSON.stringify(result);
+    process.env.FIREBASE_PROJECT = firebaseConfig;
+    process.env.DB_NAMESPACE = options.instance;
```
Comment (mbleigh) [inline lib/serve/functions.js]: Again, why `DB_NAMESPACE`?

File: lib/parseTriggersEmulator.js (hunk 1 above)
Comment (laurenzlong) [inline lib/parseTriggersEmulator.js]: GCLOUD_PROJECT currently already exists in the GCF runtime. It is used by each provider when constructing the builder, eg: https://github.com/firebase/firebase-functions/blob/master/src/providers/auth.ts#L31

Comment (laurenzlong) [inline lib/parseTriggersEmulator.js]: FIREBASE_PROJECT doesn't exist in the runtime yet, so for now it is just used for the purpose of emulating functions.config().firebase

File: lib/serve/functions.js (hunk 2 above)
Comment (laurenzlong) [inline lib/serve/functions.js]: Good point, DB_NAMESPACE used to be needed by the SDK during trigger parsing, it is no longer needed. I've removed this from both the serve and deploy process.

**Comment by laurenzlong** [comment]
Also, I can't figure out why the build is broken. Any clues? 
screenshot from 2017-05-10 12 19 28

**Comment by mbleigh** [comment]
@laurenzlong looks like there's a lint error in the branch currently.

**Comment by laurenzlong** [comment]
@mbleigh  Fixed!

**Pull request merged**